.github/workflows/pull_code_delete_patch.yml

```diff
@@ -0,0 +1,113 @@
+name: Pull Upstream and Remove Merged Downstream Patches
+
+on:
+  workflow_dispatch:
+    inputs:
+      upstream_repo:
+        description: 'Select Upstream Repository'
+        required: true
+        type: choice
+        options:
+          - 'openvino_model_server'
+          - 'kserve'
+          - 'modelmesh'
+          - 'caikit-tgis-serving'
+          - 'openvino'
+          - 'vllm'
+          - 'caikit-nlp'
+          - 'caikit'
+          - 'odh-model-controller'
+          - 'caikit-tgis-backend'
+          - 'caikit-nlp-client'
+          - 'model-registry'
+      upstream_branch:
+        description: 'Upstream branch to pull from'
+        required: true
+      target_branch:
+        description: 'Target branch to pull into'
+        required: true
+      commit_sha:
+        description: 'The merge commit SHA to delete'
+        required: true
+      action_type:
+        description: 'Specify if the changes should be pushed directly or through a pull request'
+        required: true
+        default: 'pr'
+        type: choice
+        options:
+          - 'pr'
+          - 'push'
+permissions:
+  contents: write
+  packages: write
+  pull-requests: write
+
+jobs:
+  pull_and_clean:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set repository
+        id: set-repo
+        run: |
+          echo "upstream_org_repo=opendatahub-io/${{ github.event.inputs.upstream_repo }}" >> $GITHUB_OUTPUT
+          echo "target_org_repo=red-hat-data-services/${{ github.event.inputs.upstream_repo }}" >> $GITHUB_OUTPUT
```

Comment:
Same feedback, upstream is the community, midstream is opendatahub-io, we'd be dropping the patch from midstream and incorporating the one from upstream